docs/index.adoc

```diff
@@ -218,139 +218,180 @@ libStatic.buildGetter('my/folder', {
++
+NOTE: XP version *7.7.1* is the first version where these issues are handled well. On earlier versions, trailing slashes are stripped from `.rawPath` too, so index fallback can't be expected to work. Upgrade XP if necessary.
+
+The next examples show how to achieve this in contexts outside of services:
+
+{zwsp} +
+
+=== 5. XP controller mapping
+
+Using <<#example-getcleanpath, .getCleanPath>>, lib-static can be used to serve assets from link:https://developer.enonic.com/docs/xp/stable/cms/mappings[mapped controllers].
+
+This example uses regular expressions to support the `.getCleanPath` <<#example-gotchas, criteria>>, and will serve assets (including index fallbacks) from the root _my/folder_ on the endpoint `<siteUrl>/static`:
+
+.src/main/resources/site/site.xml:
+[source,xml,options="nowrap"]
+----
+<mapping controller="/controllers/static.js" order="50">
+    <pattern>/static(/.*)?$</pattern>
+</mapping>
 ----
-exports.get = libStatic.buildGetter({
-    root: 'my/folder',
-    getCleanPath: function(request) {
-        return request.params.filename + ".txt";
-    }
 
-    // Request: **/this/is/an/endpoint?filename=myFile
-    // --> Relative resource path: myFile.txt
-    // --> lib-static looks up my/folder/myFile.txt
-});
+.src/main/resources/controllers/static.js:
+[source,javascript,options="nowrap"]
 ----
+const getStatic = libStatic.buildGetter({
+        root: `my/folder`,
+        getCleanPath: request => {
+            const basePath = `${libPortal.getSite()._path}/static`;
+            const pattern = new RegExp(`${basePath}(/.*)?$`);
+            const matched = (request.rawPath || '').match(pattern);
+
+            if (!matched) {
+                throw Error(`basePath ($basePath}) was not found in request.rawPath (${request.rawPath}`);
+            }
 
-...etc.
+            return matched[1] || '';
+        },
+    });
 
-👉 link:#options[Options API reference]
+exports.get = request => getStatic(request);
+----
 
-{zwsp} +
 
+{zwsp} +
 
 [[example-webapp]]
-=== 5. A webapp with lib-router
+=== 6. A webapp with lib-router
 
-Combining `.getCleanPath` with link:https://developer.enonic.com/docs/router-library/master[lib-router] can be an easy alternative to setting up separate services the way we did above. Just let the webapp itself use lib-router to detect sub-URI's and handle the resource serving too, all from the same controller:
+Lib-static can also be used to serve assets on URIs directly below an link:https://developer.enonic.com/docs/xp/stable/runtime/engines/webapp-engine#webapp_js[XP webapp].
 
+The easiest way is to use link:https://market.enonic.com/vendors/enonic/router-lib[the Router lib] to handle sub-URIs and extract a path relative to the webapp (`libRouterPath`), and let <<#example-getcleanpath, .getCleanPath>> fetch that from the request and return it - automatically handling the `.getCleanPath` <<#example-gotchas, gotchas>>.
```

Comment:
> _automatically handling the .getCleanPath gotchas_...

This is not true at the moment, because of the lib-router root-pattern thing. So the best approach seems to be to get lib-router fixed asap, and wait with releasing this entire doc update PR until we have a version that will work.

Unless you have reason to believe lib-router won't be fixed anytime soon, @rymsha? In that case I'll need to rewrite some stuff here so that we have some doc update to release.

Reply:
https://github.com/enonic/lib-router/pull/38 should fix the problem